src/main/scala/devices/spi/TLSPIFlash.scala

```diff
@@ -30,14 +30,14 @@ case class SPIFlashParams(
     csWidth: Int = 1,
     delayBits: Int = 8,
     divisorBits: Int = 12,
-    sampleDelay: Int = 2)
+    fineDelaySelectWidth: Int = 5,
+    sampleDelayRegSize: Int = 5)
   extends SPIFlashParamsBase {
   val frameBits = 8
   val insnAddrBytes = 4
   val insnPadLenBits = 4
 
   require(insnPadLenBits <= delayBits)
-  require(sampleDelay >= 0)
```

Comment:
Should we require this for your new parameters?

Reply:
It is not needed. The only thing would be that sample delay should be equal to or more than the delays present outside (to prevent early sampling).

Reply:
done.